src/tree.js

```diff
@@ -79,6 +80,15 @@ export function createInternal(vnode, parentInternal) {
 		_depth: parentInternal ? parentInternal._depth + 1 : 0
 	};
 
+	if (internal._flags & TYPE_ELEMENT && type === 'svg') {
```

Comment:
I wonder if the slight slowdown in the `replace1k` benchmark is caused by these checks being called for every internal, including text nodes. My assumption is that we don't need that information for text nodes and can move these checks to the branch above where we do the flag detection logic.

Reply:
Updated so these only happen in the non-text node branch. Had to move the application of inherited modes higher up in this method though since these SVG checks (specifically the `foreignObject` one) need to happen after inherited modes have been applied (the `foreignObject` mode effectively turns off the inherited MODE_SVG flag).

In my local testing this didn't increase the size much. And I think I like it this way better cuz it explicitly enables us to modify MODE inheritance in special cases